File: server/provider/schedule.go
package provider

import "time"

type Lesson struct {
	Name    string
	Cabinet string
}

type DaySchedule map[string][]Lesson
type Schedule []DaySchedule

type ScheduleStatus struct {
	Parsed time.Time `json:"parsed"`
	Hash   string    `json:"hash"`
}

type ScheduleProvider interface {
	Day(day time.Time) (*DaySchedule, error)
	Week() (*Schedule, error)
	Status() ScheduleStatus
}

File: sp_server/schedule.py
"""Типы расписания."""

from datetime import datetime, time

from pydantic import BaseModel


class Status(BaseModel):
    """Статус расписания."""

    parsed: datetime
    hash: str
    url: str


class Lesson(BaseModel):
    """Урок в расписании."""

    lesson: str
    cabinet: str


class LessonTime(BaseModel):
    """Время урока."""

    start: tuple[int, int]
    end: tuple[int, int]


DayLessons = list[Lesson]
WeekLessons = list[DayLessons]
Schedule = dict[str, WeekLessons]
TimeTable = list[LessonTime]
